pkg/staticpod/staticpod.go

```diff
@@ -231,6 +269,50 @@ func addVolumes(p *v1.Pod, src []string, dir bool) {
 	}
 }
 
+func addExtraMounts(p *v1.Pod, extraMounts []string) {
+	var (
+		prefix     = "extra-mount"
+		sourceType = v1.HostPathDirectoryOrCreate
+	)
+
+	for i, rawMount := range extraMounts {
+		mount := strings.Split(rawMount, ":")
+		if len(mount) != 2 {
+			logrus.Errorf("mount for pod %s %s was not valid", p.Name, rawMount)
+			continue
+		}
+		name := fmt.Sprintf("%s-%d", prefix, i)
+		p.Spec.Volumes = append(p.Spec.Volumes, v1.Volume{
+			Name: name,
+			VolumeSource: v1.VolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Path: mount[0],
+					Type: &sourceType,
+				},
+			},
+		})
+		p.Spec.Containers[0].VolumeMounts = append(p.Spec.Containers[0].VolumeMounts, v1.VolumeMount{
+			Name:      name,
+			ReadOnly:  false,
+			MountPath: mount[1],
+		})
+	}
+}
+
+func addExtraEnv(p *v1.Pod, extraEnv []string) {
+	for _, rawEnv := range extraEnv {
+		env := strings.Split(rawEnv, "=")
```

Comment:
I think this would preclude setting environment variables with an = in them as you'll end up with too many parts.
```suggestion
		env := strings.Split(rawEnv, "=", 2)
```